rusk/src/lib/chain/rusk.rs

```diff
@@ -569,6 +576,11 @@ fn reward_slash_and_update_root(
         reward = dusk_value
     );
 
+    let reward_per_quota = generator_extra_reward / (21 * 2);
+    let generator_curr_extra_reward =
+        credits.saturating_sub(43 * 2) * reward_per_quota;
+
+    let generator_reward = generator_fixed_reward + generator_curr_extra_reward;
```

Comment:
Can you add some tests?

Maybe I'm wrong, but I suspect we'll have some weird `generator_reward` in case of odd `dusk_spent`

Reply:
What should be a valid test to reproduce a weird generator_reward?

Reply:
Basically what I mean is that `calc_generator_extra_reward` is currently rounding down. So that if we have an extra reward of 100 and we divide be 64, the result is 1.

Are we sure we want to burn that amount too?

Reply:
Added.